src/main/java/org/folio/rest/impl/HoldingsStorageAPI.java

```diff
@@ -316,33 +329,43 @@ public void putHoldingsStorageHoldingsByHoldingsRecordId(
           postgresClient.get(HOLDINGS_RECORD_TABLE, HoldingsRecord.class, fieldList, cql, true, false,
             reply -> {
               if(reply.succeeded()) {
-                List<HoldingsRecord> itemList = reply.result().getResults();
+                List<HoldingsRecord> holdingsList = reply.result().getResults();
 
-                if (itemList.size() == 1) {
+                if (holdingsList.size() == 1) {
                   try {
-                    postgresClient.update(HOLDINGS_RECORD_TABLE, entity, entity.getId(),
-                      update -> {
-                        try {
-                          if (update.succeeded()) {
-                            asyncResultHandler.handle(
-                              Future.succeededFuture(
-                                PutHoldingsStorageHoldingsByHoldingsRecordIdResponse
-                                  .respond204()));
-                          }
-                          else {
-                            asyncResultHandler.handle(
-                              Future.succeededFuture(
-                                PutHoldingsStorageHoldingsByHoldingsRecordIdResponse
-                                  .respond500WithTextPlain(
-                                    update.cause().getMessage())));
-                          }
-                        } catch (Exception e) {
-                          asyncResultHandler.handle(
-                            Future.succeededFuture(
-                              PutHoldingsStorageHoldingsByHoldingsRecordIdResponse
-                                .respond500WithTextPlain(e.getMessage())));
-                        }
+                    postgresClient.startTx(connection -> {
+                      updateItemEffectiveCallNumbersByHoldings(connection, postgresClient, entity).thenAccept(voidResult -> {
```

Comment:
We need to rollback if `updateItemEffectiveCallNumbersByHoldings` fails. If it fails, the `thenAccept` consumer won't be called. If we continue to use `CompletableFuture`, we would need an `exceptionally` block that could roll back on error. If we use a Vert.x `Future`, we could use `setHandler` to determine if the call number update succeeded or failed and apply the `postgresClient` update or fail.